src/writer.rs

```diff
@@ -149,51 +212,61 @@ where
 /// Write the data in 'f' order.
 ///
 /// Like NiBabel, we iterate by "slice" to improve speed and use less memory.
-fn write_data<A, S, D, W>(writer: &mut W, header: NiftiHeader, data: &ArrayBase<S, D>) -> Result<()>
+fn write_data<T, D, W>(writer: &mut W, header: NiftiHeader, data: ArrayView<T, D>) -> Result<()>
```

Comment:
`header` can be passed by reference, since it's not meant to be consumed here.

Reply:
Hum, ok? I did it, no problem, but can you please explain why this should be done? I had a "pragmatic" view: "I don't need it anymore, lets pass it by value", but this may not be idiomatic to Rust. I had some problems in other projects where some callers want to pass by value and others by reference, so I didn't know what to do.

By "it's not meant to be consumed", there's clearly an intent. If the function is calling `into_*` or an object's life should clearly end there, it should be pass by value?

Reply:
Long answer: This isn't a hard rule, since it's a private function and will hardly influence how the program works. My perception of this is that a reference to the header is reasonable from both ends of the spectrum: on the implementation side, the function does need to own any of its parts, nor does it seem that it could take advantage of owning the object. On the interface side, the header provides a piece of processed information to produce an output through a writer channel, and so will have to replicate it into the byte stream regardless. As far as I know in terms of API conventions, the fact that it's a terminal operation doesn't mean that it should consume its operands. As an example, encoding functions in serde usually receive the serializable object by reference.

The short answer is that Clippy has a default lint for this. 😅 The lint itself, `needless_pass_by_value`, has a known history of lame false positives, but here a trigger is reasonable (for the reasons above).